asserts/database.go

```diff
@@ -255,11 +256,24 @@ func (db *Database) Add(assert Assertion) error {
 	if err != nil {
 		return err
 	}
-	for _, k := range assertType.PrimaryKey {
-		if assert.Header(k) == "" {
+
+	keyValues := make([]string, len(assertType.PrimaryKey))
+	for i, k := range assertType.PrimaryKey {
+		keyVal := assert.Header(k)
+		if keyVal == "" {
 			return fmt.Errorf("missing primary key header: %v", k)
 		}
+		keyValues[i] = keyVal
+	}
+
+	// assuming trusted account keys/assertions will be managed
+	// through the os snap this seems the safest policy until we
+	// know more/better
+	_, err = db.trusted.Get(assertType, keyValues)
+	if err != ErrNotFound {
+		return fmt.Errorf("cannot add assertion of type %q with primary key clashing with a trusted assertion: %v", assertType.Name, keyValues)
```

Comment:
I'd simplify the first part of this message to, "cannot add %q assertion with primary ...".